services/cd-service/pkg/service/overview.go

```diff
@@ -147,6 +150,27 @@ func (o *OverviewServiceServer) getOverview(
 							DeployTime:   "",
 						},
 					}
+					//we don't want to insert the locks for a team twice, so init a set of checked teams and filter the ones we already processed.
+					teamName, err := s.GetTeamName(appName)
+					_, exists := teamSet[teamName]
+					if err == nil && !exists {
+						teamSet[teamName] = true
+						if teamLocks, teamErr := s.GetEnvironmentTeamLocks(envName, teamName); teamErr == nil {
```

Comment:
What if there is a `teamErr`?
It's reasonable to still return something to the UI here, but we should at least log a warning.